icu4c/source/common/brkiter.cpp

```diff
@@ -438,13 +438,12 @@ BreakIterator::makeInstance(const Locale& loc, int32_t kind, UErrorCode& status)
             if (U_SUCCESS(kvStatus) && (value == "strict" || value == "normal" || value == "loose")) {
                 uprv_strcat(lbType, "_");
                 uprv_strcat(lbType, value.data());
-            } else {
-                value.clear();
-                loc.getKeywordValue("lw", valueSink, kvStatus);
-                if (U_SUCCESS(kvStatus) && value == "phrase") {
-                    uprv_strcat(lbType, "_");
-                    uprv_strcat(lbType, value.data());
-                }
+            }
+            value.clear();
```

Comment:
Notice value is used in the line
```
            UTRACE_DATA1(UTRACE_INFO, "lb=%s", value.data());
```
later

change 
```
CharString value;
CharStringByteSink valueSink(&value)
```
to
```
CharString lbValue;
CharStringByteSink lbValueSink(&lbValue);
CharString lwValue;
CharStringByteSink lwValueSink(&lwValue)
```
and use lbValue and lwValue for the two if check.
Also change 
```
UTRACE_DATA1(UTRACE_INFO, "lb=%s", value.data());
```
to
```
UTRACE_DATA1(UTRACE_INFO, "lb=%s lw=%s", lbValue.data(), lwValue.data());
```

You will then need to change and add test in third_party/icu/source/test/intltest/rbbitst.cpp . 
You will need to change traceData and assertTestTraceResult in that file to verify both lb value and lw value. 
You should also add similar test of TestTraceCreateLine / TestTraceCreateLineStrict / TestTraceCreateLineNormal / TestTraceCreateLineLoose

Reply:
Since lbType concatenates lb and lw, use it to verify.